src/migrations/20230802000000-genericReport.js

```diff
@@ -0,0 +1,2853 @@
+      await queryInterface.createTable('Foiaable', {
+        id: {
+          type: Sequelize.INTEGER,
+          allowNull: false,
+          primaryKey: true,
+          autoIncrement: true,
+        },
+        table: {
+          type: Sequelize.TEXT,
+          allowNull: false,
+        },
+        column: {
+          type: Sequelize.TEXT,
+          allowNull: false,
+        },
+        createdAt: {
+          type: Sequelize.DATE,
+          allowNull: false,
+          defaultValue: Sequelize.fn('NOW'),
+        },
+        updatedAt: {
+          type: Sequelize.DATE,
+          allowNull: false,
+          defaultValue: Sequelize.fn('NOW'),
+        },
+      }, { transaction });
+      //---------------------------------------------------------------------------------
+      await queryInterface.createTable('ValidFor', {
+        id: {
+          type: Sequelize.INTEGER,
+          allowNull: false,
+          primaryKey: true,
+          autoIncrement: true,
+        },
+        name: {
+          type: Sequelize.TEXT,
+          allowNull: false,
+        },
+        isReport: {
+          type: Sequelize.BOOLEAN,
+          allowNull: false,
+          defaultValue: false,
+        },
+        createdAt: {
+          type: Sequelize.DATE,
+          allowNull: false,
+          defaultValue: Sequelize.fn('NOW'),
+        },
+        updatedAt: {
+          type: Sequelize.DATE,
+          allowNull: false,
+          defaultValue: Sequelize.fn('NOW'),
+        },
+        deletedAt: {
+          type: Sequelize.DATE,
+          allowNull: true,
+          default: null,
+        },
+        mapsTo: {
+          type: Sequelize.INTEGER,
+          allowNull: true,
+          default: null,
+          onUpdate: 'CASCADE',
+          onDelete: 'CASCADE',
+          references: {
+            model: {
+              tableName: 'ValidFor',
+            },
+            key: 'id',
+          },
+        },
+      }, { transaction });
+
+      await queryInterface.sequelize.query(`
+        INSERT INTO "ValidFor"
+        ("name", "isReport", "createdAt", "updatedAt")
+        SELECT
+          t.name,
+          t.name in (${Object.values(REPORT_TYPE).map((type) => `'${type}'`).join(',\n')}),
+          current_timestamp,
+          current_timestamp
+        FROM UNNEST(ARRAY[
+          ${Object.values(ENTITY_TYPE).map((type) => `'${type}'`).join(',\n')}
+        ]) t(name)
+       ;`, { transaction });
+
+      //---------------------------------------------------------------------------------
+
+      await queryInterface.createTable('Statuses', {
+        id: {
+          type: Sequelize.INTEGER,
+          allowNull: false,
+          primaryKey: true,
+          autoIncrement: true,
+        },
+        name: {
+          type: Sequelize.TEXT,
+          allowNull: false,
+        },
+        isTerminal: {
+          type: Sequelize.BOOLEAN,
+          allowNull: false,
+          default: false,
+        },
+        ordinal: {
```

Comment:
what does ordinal mean in a case like "suspended"?

Reply:
for a set of statuses ( same valid for) ordinal defines the order of progression of the statuses.
example:
"draft" 1
"not started" 2
"in progress" 3
"suspended" 4
"complete" 5

Reply:
suspended isn't part of the progression, that's why I asked. In the UI, suspended is the final option. Trying to get at how we're using "ordinal"

Reply:
Ok, so suspended needs to be last, We can make that change.